include/lp_hash_table.h

```diff
@@ -7,6 +7,14 @@
 #include <stdlib.h>
 #include <type_traits>
 
+/*
+ * LpHashTable is a struct implementing a `linear probing hash table`.
+ * You can find more information about about that here: https://en.wikipedia.org/wiki/Linear_probing.
+ * It functions like a normal hash table but it handles collisions by linearly
+ * iterating over the hash table until an empty spot is found. The search is performed
+ * in a similar fashion. Because of the use case we are covering in the DIL compiler
+ * there is no need for a `delete` method and the size of the table is constant.
+ */
```

Comment:
Let me help in that a little. I think a good way to provide doc is to first of all split in the usage info and the internal / implementation info. That is, if you open the file, you should immediately be able to answer the questions: "What is this?", "How do I use it?", "How fast it is?". So, we may start as:
```
This is a map of const char * to pointer values (of any type) supporting insertion and search.
```
Notice that I didn't refer to a "hash table" because the fact that this is implemented as a hash table is not important to the user. Continuing...
```
Important details:
-- The number of elements to be saved is constant and is provided in its construction.
-- Keys: The contents of the memory pointed by the const char * passed are _not_ considered.
         Only its (pointer) value. That is, these pointers are handled as integer keys.
-- Values: Should be pointers to objects of any type.
Performance:
Insertion: O(1) amortized.
Search: O(1) amortized.
```
If you want to improve that, please do! Then, we'll continue to the implementation details.

Reply:
I misinterpreted what you meant by documentation. I thought that because it was source code documentation it should document the implementation rather than the higher level abstraction. I will work on it.

Reply:
Both are part of the documentation. :)
Sure, you may put them in different places depending on the code though.